haproxy_stats.sh

```diff
@@ -27,46 +18,11 @@ debug() {
 debug "SOCAT_BIN        => $SOCAT_BIN"
 debug "CACHE_FILEPATH   => $CACHE_FILEPATH"
 debug "CACHE_EXPIRATION => $CACHE_EXPIRATION minutes"
-debug "HAPROXY_SOCKET   => $HAPROXY_SOCKET"
+debug "HAPROXY_SOCK   => $HAPROXY_SOCK"
 debug "pxname   => $pxname"
 debug "svname   => $svname"
 debug "stat     => $stat"
 
-# check if socat is available in path
-if [ -z "$SOCAT_BIN" ] && [ "$GET_STATS" -eq 1 ]
-then
-  echo 'ERROR: cannot find socat binary'
-  exit 126
-fi
-
-# if we are getting stats:
-#   check if we can write to stats cache file, if it exists
-#     or cache file path, if it does not exist
-#   check if HAPROXY socket is writable
-# if we are NOT getting stats:
-#   check if we can read the stats cache file
-if [ "$GET_STATS" -eq 1 ]
-then
-  if [ -e "$CACHE_FILEPATH" ] && [ ! -w "$CACHE_FILEPATH" ]
-  then
-    echo 'ERROR: stats cache file exists, but is not writable'
-    exit 126
-  elif [ ! -w ${CACHE_FILEPATH%/*} ]
-  then
-    echo 'ERROR: stats cache file path is not writable'
-    exit 126
-  fi
-  if [ ! -w $HAPROXY_SOCKET ]
-  then
-    echo "ERROR: haproxy socker is not writable"
-    exit 126
-  fi
-elif [ ! -r "$CACHE_FILEPATH" ]
-then
-  echo 'ERROR: cannot read stats cache file'
-  exit 126
-fi
-
```

Comment:
so we can just do this:
```
SOCAT_BIN="$(which socat)"
[ -z "$SOCAT_BIN" ] && echo ERROR: socat binary not found && exit 1
```

or, if we go the "function" route, we put this in config file:
```
check_socat() {
  SOCAT_BIN="${SOCAT_BIN:-$(which socat)}"
  [ -z "$SOCAT_BIN" ] && return 1
}
```
and do `check_socat` in whichever script needs it..

----
or even
```
socat() {
  local _socat_bin="${SOCAT_BIN:-$(which socat)}"
  [ -z "${_socat_bin}" ] && return 1
  ${_socat_bin} $@
}
```
and in whichever script, use this function, as if it's a binary